### ISSUE TITLE
VR support

### ISSUE THREAD

**Issue by yaomtc** [drwhut/tabletop-club#285]
**Is your feature request related to a problem? Please describe.**
[Godot supports OpenXR], and that proprietary simulator supports VR. You mention wanting to support more platforms. VR could be a great future target! OpenXR can be handled with the open source [Monado] or the proprietary [SteamVR].

**Describe the solution you'd like**
It would be nice to be able to have full motion controller support with in-game hands, such as with my Valve Index (the only VR system "officially" supported on Linux besides the Vive)

**Describe alternatives you've considered**
I think if VR support were to be put on the roadmap, #100 should be done first. Then you can playing with a VR headset and a controller, at least. Test for rendering issues / performance when rendering for a stereo view. If that works out, maybe motion controls later.

Currently there are very few VR games/apps specifically [supporting Linux]. The vast majority only target Windows, then [Proton] is used to translate, though it [can work quite well.](https://gitlab.com/vr-on-linux/VR-on-Linux/-/tree/master/Proton)

[Godot supports OpenXR]: https://docs.godotengine.org/en/stable/tutorials/xr/setting_up_xr.html
[Monado]: https://monado.dev/
[SteamVR]: https://github.com/ValveSoftware/SteamVR-for-Linux
[Proton]: https://github.com/ValveSoftware/Proton
[supporting Linux]: https://gitlab.com/vr-on-linux/VR-on-Linux

**Comment by drwhut** [comment]
I agree, having VR support for Tabletop Club would be awesome! 😁 However, I myself will not be able to add support to the game, since I do not own a VR headset with controllers (unless there is a way to simulate VR input, e.g. headset and controller position, but I am not aware of anything like this).

But that does not stop anyone that DOES have a headset from contributing! In the `0.2-rewrite` branch I'm planning on making the camera controller a lot more modular so it's much easier to implement control schemes for different platforms (e.g. mobile, VR).
